### PR TITLE
Apply rules test

### PR DESCRIPTION
**Pull request by ryancheta**
My first PR. Don't haze the n00b please. 👍🏾

### CODE REVIEW

File: services/rules-faas/tests/applyRules.test.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    {
+    name: 'Paper Towels',
+    price: '12.3348',
+    quantity: '1',
+    author: 'Joe Namath',
+    debitor: 'Joe Smith',
+    creditor: 'Mary',
+    },
+  
+    // {
```
Comment (mxfactorial) [inline services/rules-faas/tests/applyRules.test.js]: remove unused items

File: services/rules-faas/tests/applyRules.test.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        with_sales_tax.forEach( item => { 
+            if(item.name !== TAX_TRANSACTION_NAME){
+                expect(Object.keys(item).length).toEqual(length)
+            } 
+        })
+    })
+
+    it('decimal precision is fixed to thousandths', () => {
+        let isOK = false
+        console.log(with_sales_tax)
```
Comment (mxfactorial) [inline services/rules-faas/tests/applyRules.test.js]: remove logging

File: services/rules-faas/tests/applyRules.test.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        })
+    })
+    
+    it('checks debitor name match', () => {
+        let items = with_sales_tax.filter( item => {
+            return item.name !== TAX_TRANSACTION_NAME
+        });
+        
+        for(i = 0; i < items.length; i++){
+            expect(items[i].author ).toEqual(transaction_items[i].author);
```
Comment (mxfactorial) [inline services/rules-faas/tests/applyRules.test.js]: use [asi](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Automatic_semicolon_insertion); remove semicolon

also, remove space after expect argument.

File: services/rules-faas/tests/applyRules.test.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    //   name: '9% Sales Tax',
+    //   price: '7.02',
+    //   quantity: '10',
+    //   author: 'Joe Smith',
+    //   debitor: 'Joe Smith',
+    //   creditor: 'StateOfCalifornia'
+    // }
+]
+
+const with_sales_tax = applyRules( uuidv1(), transaction_items )
```
Comment (mxfactorial) [inline services/rules-faas/tests/applyRules.test.js]: spaces **between** arguments only.
remove space before initial argument and after last argument

**Review by mxfactorial — CHANGES_REQUESTED** [review]

File: services/rules-faas/tests/applyRules.test.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        
+        with_sales_tax.forEach( item => { 
+            if(item.name !== TAX_TRANSACTION_NAME){
+                expect(Object.keys(item).length).toEqual(length)
+            } 
+        })
+    })
+
+    it('decimal precision is fixed to thousandths', () => {
+        let isOK = false
```
Comment (mxfactorial) [inline services/rules-faas/tests/applyRules.test.js]: not used; removed

**Review by mxfactorial — CHANGES_REQUESTED** [review]

File: services/rules-faas/tests/applyRules.test.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    {
+      name: 'Grapes',
+      price: '5.223',
+      quantity: '1',
+      author: 'Joe Namath',
+      debitor: 'Joe Smith',
+      creditor: 'Mary'
+    },
+    {
+    name: 'Paper Towels',
```
Comment (oozywaters) [inline services/rules-faas/tests/applyRules.test.js:48]: Add indentation please

File: services/rules-faas/tests/applyRules.test.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+            if(item.name !== TAX_TRANSACTION_NAME){
+                expect(Object.keys(item).length).toEqual(length)
+            } 
+        })
+    })
+
+    it('decimal precision is fixed to thousandths', () => {
+        with_sales_tax.forEach( item => {
+            if( item.name == TAX_TRANSACTION_NAME ){
+                expect( item.price ).toEqual( parseFloat(item.price).toFixed(3) )
```
Comment (oozywaters) [inline services/rules-faas/tests/applyRules.test.js]: redundant spaces around function arguments

**Comment by oozywaters** [comment]
Please, set up eslint in your IDE to avoid code styling issues

**Pull request merged**